simulation/src/node.ts

```diff
@@ -1,364 +1,119 @@
-import {
-  AVERAGE_COMPUTE,
-  AVERAGE_CRYPTO,
-  HEALTH_CHECK_PERIOD,
-  MAX_LATENCY,
-  MULTICAST_SIZE
-} from './manager'
 import { Message, MessageType } from './message'
-import { Generator } from './random'
+import {
+  handleContributionTimeout,
+  handleRequestContribution,
+  handleSendContribution,
+  handleConfirmContributors,
+  handleShareContributors,
+  handleSendAggregate,
+  handleRequestHealthChecks,
+  handleHealthCheckTimeout,
+  handleContactBackup,
+  handleBackupResponse,
+  handleConfirmBackup,
+  handleSendChildren,
+  handleRequestData
+} from './messageHandlers'
 import TreeNode from './treeNode'
 
-const BASE_NOISE = 10000000
+export const arrayEquals = (a: number[], b: number[]): boolean => {
+  return JSON.stringify(a.sort()) === JSON.stringify(b.sort())
```

Comment:
lodash has a lot of useful helpers. Here you could use https://lodash.com/docs/4.17.15#isEqual

Reply:
Also, an `arrayEquals` function used elsewhere would be more relevant in a `utils` file rather than in this big `node` file.